app/bulk_resource/views.py

```diff
@@ -92,10 +93,95 @@ def upload_data():
     db.session.commit()
     # TODO: If the only descriptors are the required ones, then no need to
     # TODO: review them.
+    reset_url = url_for('bulk_resource.review_descriptor_types')
+    update_url = url_for('bulk_resource.update_data')
+    redirect_url = reset_url if reset_or_update == 'reset' else update_url
     return jsonify(
-        redirect=url_for('bulk_resource.review_descriptor_types')
+        redirect=redirect_url
     )
 
+@bulk_resource.route('/update-data', methods=['GET'])
+@login_required
+def update_data():
+    csv_container = CsvContainer.most_recent(user=current_user)
+    if csv_container is None:
+        abort(404)
+
+    errors = []
+
+    for row in csv_container.csv_rows:
+        name = row.csv_body_cells[csv_container.name_column_index].data
+        address = \
+            row.csv_body_cells[csv_container.address_column_index].data
+        g = geocoder.google(address)
+        resource = Resource.query.filter_by(name=name).first()
+
+        if resource is None:
+            resource = Resource(
+                name=name,
+                address=address,
+                latitude=g.latlng[0],
+                longitude=g.latlng[1]
+            )
+            db.session.add(resource)
+
+        else:
+            resource.latitude = g.latlng[0]
+            resource.longitude = g.latlng[1]
+
+        for i, cell in enumerate(row.csv_body_cells):
+            if i not in csv_container.required_column_indices():
+                descriptor_name = \
+                    csv_container.csv_header_row.csv_header_cells[i].data
+                descriptor = Descriptor.query.filter_by(
+                    name=descriptor_name
+                ).first()
+                if descriptor is None:
+                    errors.append('Descriptor {} is invalid'.format(descriptor_name))
+                    continue
+
+                text_association = [td for td in resource.text_descriptors if td.descriptor_id == descriptor.id]
+                text_association = text_association[0] if text_association else None
+                option_association = [od for od in resource.option_descriptors if od.descriptor_id == descriptor.id]
+                option_association = option_association[0] if option_association else None
+
+                if text_association is not None:
+                    db.session.delete(text_association)
+                elif option_association is not None:
+                    db.session.delete(option_association)
```

Comment:
Correct me if I'm wrong but I'm assuming what you're doing is first deleting the option associations for the resource and then creating new associations right? We now support multiple option associations so instead of deleting `option_association[0]` you would have to delete all